packages/intuit/src/Utils.js

```diff
@@ -0,0 +1,45 @@
+import { composeNextState } from '@openfn/language-common';
+import {
+  request as commonRequest,
+  logResponse,
+} from '@openfn/language-common/util';
+
+export const prepareNextState = (state, response) => {
+  const { body, ...responseWithoutBody } = response;
+
+  if (!state.references) {
+    state.references = [];
+  }
+
+  return {
+    ...composeNextState(state, response.body),
+    response: responseWithoutBody,
+  };
+};
+
+export const request = (state, method, path, options = {}) => {
+  const { baseUrl, access_token } = state.configuration;
+
+  const {
+    query = {},
+    body = {},
+    headers = {
+      'Content-type': 'application/json',
```

Comment:
hmm I think you've done this the wrong way around?

If a user passes their own headers object, then these defaults won't be applied. It's an edge case but it might just catch someone out one day.

Can you move the defaults down into Opts, alongside Authorization?